doc/tutorials/changetracking.md

```diff
@@ -0,0 +1,112 @@
+# Changetracking
+
+Elektra provides developers of applications and plugins a way to determine changes made to a `KeySet` relative to the last known state of the key database.
+This can be useful if you are writing a plugin that is only supposed to working with changed data.
+
+## Basics
+
+The two headers you have to use are `kdbchangetracking.h` and `kdbdiff.h`.
+Those declare the `elektraChangeTracking*` and `elektraDiff*` methods.
+
+The two main data structures you will encounter are `ChangeTrackingContext` and `ElektraDiff`.
+
+## Getting the difference between KeySets
+
+If all you want to do is get the difference between two in-memory `KeySet` objects, use the function `elektraDiffCalculate`.
+
+```c
+KeySet * originalKeySet;
+KeySet * modifiedKeySet;
+Key * parentKey;
+
+// Calculate the difference of the keys below and including parentKey
+ElektraDiff * diff = elektraDiffCalculate (modifiedKeySet, originalKeySet, parentKey);
+
+// Extract useful information, see section 'Working with the diff'
```

Comment:
Forward references should be avoided in scientific texts but okay, to keep the funny style here, simply write something like "Don't be impatient, we will soon look into ..." :wink:

Reply:
Good that it's not a scientific text then 😃